main.nf

```diff
@@ -19,4 +30,39 @@ workflow mtbseq {
 
 }
 
+workflow per_sample {
+    reads_ch = Channel.fromFilePairs(params.reads)
+
+    TBBWA(reads_ch, params.gatk38_jar, params.user)
+    TBREFINE(TBBWA.out.next_step, params.gatk38_jar, params.user)
+    TBPILE(TBREFINE.out.next_step, params.gatk38_jar, params.user)
+    TBLIST(TBPILE.out.next_step, params.gatk38_jar, params.user)
+    TBVARIANTS(TBLIST.out.next_step, params.gatk38_jar, params.user)
+    TBSTATS(TBBWA.out.next_step,TBLIST.out.next_step, params.gatk38_jar, params.user)
+    TBSTRAINS(TBLIST.out.next_step, params.gatk38_jar, params.user)
+
+
+
+}
 
+workflow cohort {
+    reads_ch = Channel.fromFilePairs(params.reads)
+
+    TBBWA(reads_ch, params.gatk38_jar, params.user)
+    TBREFINE(TBBWA.out.next_step, params.gatk38_jar, params.user)
+    TBPILE(TBREFINE.out.next_step, params.gatk38_jar, params.user)
+    TBLIST(TBPILE.out.next_step, params.gatk38_jar, params.user)
+    TBVARIANTS(TBLIST.out.next_step, params.gatk38_jar, params.user)
+    TBSTATS(TBBWA.out.next_step,TBLIST.out.next_step, params.gatk38_jar, params.user)
+    TBSTRAINS(TBLIST.out.next_step, params.gatk38_jar, params.user)
+
+    samples_tsv_file = TBBWA.out.genomes_names
+            .collect()
+            .flatten().map { n -> "$n" + "\t" + "${params.library_name}" + "\n" }
+            .collectFile(name: 'samples.tsv', newLine: false, storeDir: "${params.outdir}")
+
+    TBJOIN(samples_tsv_file,TBVARIANTS.out[0].collect(), TBLIST.out[0].collect(), params.gatk38_jar, params.user)
```

Comment:
Let's name the emitted values here as well in `TBVARIANTS/ TBLIST` etc.

Reply:
ok :+1: